api_fhir_r4/converters/medicationConverter.py

```diff
@@ -119,12 +110,10 @@ def build_fhir_package_amount(cls, fhir_medication, imis_medication):
     @classmethod
     def split_package_amount(cls, amount):
         amount = amount.lstrip()
-        if " " not in amount:
-            return None
-        if " " in amount:
-            amount = amount.split(' ', 1)
-            amount = amount[0]
-            return int(amount)
+        try:
+            return int(re.sub("[^0-9]","",amount))
+        except ValueError as exception:
+            return 0
```

Comment:
are you ok with this?

Reply:
@toch do you combine the quantity with the format without any space chars? If that's the case then it will be difficult for users to read.

Reply:
@dragos-dobre as explained in the description:

> When working with the demo database, it appeared that several medications didn't have a package complying with that format: the unit is sometimes not separated from the integer by the space (e.g. 10000TABLETS, 10ml, 200T), or there is only a text (e.g. EACH).

So, I haven't changed anything in the dataset. It contains "amount with unit not not separated by space" and "non numberical text". It causes the bug when listing all medications.

AFAIU, the demo database is a kind of reference, so I considered that it should work with those data, and made the present fix. (also I guess as it was used in the rest of the application, it was in a certain way validated)

I don't have any opinion regarding what format it should comply with. If there is a particular format to comply with, I guess the demo database should be fixed. I don't know what would be the consequence of a such change. If there isn't, it seems to me a such fix could be an option.

what should I consider?

Reply:
@dragos-dobre As Christophe pointed out, the API was crashing when listing many of the items. This is making the algorithm more robust even if it's not perfect. There is no sure way of splitting this but we can make a reasonable assumption that doesn't crash.
Can we merge this, it is quite blocking for the mobile app to FHIR porting ?